pkg/reboot/rebooter.go

```diff
@@ -82,7 +82,7 @@ func (r *watchdogRebooter) Reboot() error {
 func (r *watchdogRebooter) softwareReboot() error {
 	r.log.Info("about to try software reboot")
 	// hostPID: true and privileged:true required to run this
-	rebootCmd := exec.Command("/usr/bin/nsenter", "-m/proc/1/ns/mnt", "/bin/systemctl", "reboot", "--force", "--force")
+	rebootCmd := exec.Command("bash", "-b", "echo c > /proc/sysrq-trigger")
```

Comment:
do we need the full path to bash?

Reply:
And maybe it needs to be done inside an `nsenter` context?

 ```suggestion
	rebootCmd := exec.Command("/usr/bin/nsenter", "-m/proc/1/ns/mnt", "bash", "-b", "echo c > /proc/sysrq-trigger")
```

Reply:
>And maybe it needs to be done inside an nsenter context?

Not an expert on `nsenter` , but from the [little reading](https://man7.org/linux/man-pages/man1/nsenter.1.html) I've done my understanding is that  the  `mnt` [option](https://man7.org/linux/man-pages/man7/mount_namespaces.7.html)  would create an isolated namespace for the command to run in.

Since in any case we plan to reboot the whole machine is there a value in that separation which  I'm missing ? (my initial intuition is that it's just an unneeded overhead)

Reply:
> my understanding is that the mnt [option](https://man7.org/linux/man-pages/man7/mount_namespaces.7.html) would create an isolated namespace for the command to run in.

no, you don't create that namespace, you *enter* it. With `-m/proc/1/ns/mnt` you enter the "mount" namespace of the process with PID `1`. That's the host' s very first process. This gives you access to binaries and files which are installed on the host but not in the container you are running in, like `/bin/systemctl` in the old version, or `/bin/bash` and `/proc/sysrq-trigger` in your new version. That's what I understand at least. So while bash *might*  be installed in the container as well, I'm not sure if/how `/proc/sysrq-trigger` works in containers...